lib/KettleRequest.js

```diff
@@ -381,17 +417,26 @@ kettle.request.http.errorHandler = function (res, error) {
 /**
  * Send a successful payload to the client if the success event is fired
  * @param {Component} request - A kettle.request component
- * @param {Object} response A success payload
+ * @param {Object} responseBody The payload of a success response
+ * @param {Number} responseOptions Additional metadata for the response, e.g. statusCode
  */
-kettle.request.http.successHandler = function (request, response) {
+kettle.request.http.successHandler = function (request, responseBody, responseOptions) {
+    // Extract response options to be applied to the request
+    // TODO: do we want to react to other keys besides statusCode? mimeType, headers?
+
+    // The default status code for a successful HTTP response is 200 OK
+    var statusCode = responseOptions && responseOptions.statusCode || 200;
+    // var headers = responseOptions && responseOptions.headers || {};
+
     if (request.req.method.toLowerCase() === "options") {
-        request.res.status(200).end();
+        request.res.status(statusCode !== undefined ? statusCode : statusCode).end();
         return;
     }
```

Comment:
This is a strange little fork in the code. HTTP OPTIONS requests should be honored with a response body, so it's unclear why they are short-circuited here. Should we just remove this and see if tests still pass?

Reply:
Yes, and if they do, add another one :)

Reply:
[This test](https://github.com/fluid-project/kettle/blob/master/tests/GoodRequestTests.js#L115) fails, though it seems to exist only to verify this inexplicable behavior.